### DISCUSSION TITLE
Deployer 7.5.8 has issues Deploying Magento 2

### DISCUSSION THREAD

**Discussion by Aquive** [deployphp/deployer#3984] [Bugs]
### Deployer Version

7.5.8

### Target OS

Debian

### Which PHP version are you using?

PHP 8.2

### Content of deploy.php or deploy.yaml

```
<?php
namespace Deployer;

require 'recipe/magento2.php';

// Config
set('repository', '#');
set('update_code_strategy', 'clone');
set('writable_mode', 'chmod');
set('keep_releases', 3);

add('shared_files', []);
add('shared_dirs', []);
add('writable_dirs', []);

set('static_content_locales', 'nl_NL');
set('magento_themes', ['client/default']);
set('magento_themes_backend', ['Magento/backend']);
set('static_deploy_options', '--no-parent');
set('static_content_locales_backend', 'en_US nl_NL');
set('split_static_deployment', true);

// Hosts
host('production')
    ->set('hostname', '#')
    ->set('remote_user', 'app')
    ->set('deploy_path', '~/production')
    ->set('static_content_jobs', '2');

host('staging')
    ->set('hostname', '#')
    ->set('remote_user', 'app')
    ->set('deploy_path', '~/staging')
    ->set('static_content_jobs', '2');

// Hooks
after('deploy:failed', 'deploy:unlock');
```

### Steps to reproduce

I installed Deployer in a new project. It installed version 7.5.8. When deploying we encounter weird problems. The recipe is very basic.

- DI problems where modules do not work (404 errors on ajax routes)
- Specific errors which are resolved after manually flushing cache

I also tried adding a manual cache clear without any effect. When I checked with `-v` option I also saw it already does a cache flush.

When I downgrade to version `7.3.1` (which we successfully use in other projects) it does work without above problems.

**Comment by antonmedv** [comment]
> When deploying we encounter weird problems.

What errors? Probably those are important. I just deploying with 7.5.8 - everything works.

**Comment by Aquive** [comment]
@antonmedv thanks for your answer. Magento specific errors. Mostly code not being found (because it's actually not there/compiled). And 404 Error on ajax routes is what I encoutered so far. 

- One of the problem seems to be magento di:compile step is not succesfull somehow?
- the other one is fixed/worked around with a manual cache:flush
